tests/cpptests/test_cpp_hybridmerger.cpp

```diff
@@ -1041,3 +1048,57 @@ TEST_F(HybridMergerTest, testRRFScoring3Upstreams) {
   SearchResult_Destroy(&r);
   QITR_FreeChain(&qitr);
 }
+
+/*
+ * Test that hybrid merger correctly handles error precedence when query iterator has error set and upstreams return different states
+ *
+ * Scoring function: Hybrid linear
+ * Number of upstreams: 2
+ * Intersection: N/A (error handling test)
+ * Emptiness: Mixed (one upstream times out, one returns EOF)
+ * Timeout: Mixed (one upstream times out, one returns EOF)
+ * Expected behavior: Return RS_RESULT_ERROR (most severe error) when query iterator has error set, regardless of upstream states
+ */
+TEST_F(HybridMergerTest, testHybridMergerQueryIteratorErrorPrecedence) {
+  QueryIterator qitr = {0};
+  QueryError qerr;
+  QueryError_Init(&qerr);
+  qitr.err = &qerr;
+
+  // Set error in query iterator at the beginning
+  QueryError_SetError(&qerr, QUERY_EGENERIC, "Test error in query iterator");
+
+  // Create upstreams with different behaviors:
+  // upstream1: returns RS_RESULT_TIMEDOUT after 1 call
+  // upstream2: returns RS_RESULT_EOF (empty upstream)
+  MockUpstream upstream1(1, {1.0, 2.0}, {1, 2}); // timeoutAfterCount=1 (timeout after 1 call)
+  MockUpstream upstream2(0, {}, {}); // empty upstream (returns EOF)
```

Comment:
Missing the argument for the `errorAfterCount=0, so this will not return an error